### DISCUSSION TITLE
WEBP动画异常

### DISCUSSION THREAD

**Discussion by zoze0** [linuxdeepin/developer-center#2941] [General | 常规讨论]
![深度录屏_选择区域_20190507203411](https://user-images.githubusercontent.com/20562449/57299704-d3955600-7107-11e9-9254-1ad63fc29d8e.gif)
 左侧是在浏览器中打开,右侧是在深度看图打开.深度看图的动画速度异常

另外 深度看图也不支持APNG (animated PNG) 格式的图片,是否考虑支持一下

图片资源 : https://aotu.io/notes/2016/11/07/apng/

**Comment by dependon** [comment]
I'm sure it's a problem with qimagereader and qmovie in Qt, and the ring qt5.11.3, it hasn't been fixed

**Comment by dependon** [comment]
这个图片在qt读取的速度为0,导致了策略是无限速度一直转，可能qt不会修复该问题

**Comment by BLumia** [comment]
抱歉现在才回复 issue。

您链接中指向的 webp 图片似乎已经失效了，这边没法获取到原图来进行测试。不过，webp 是属于 Qt 官方支持的图像格式之一，如果 webp 图像显示效果有问题的话，可以汇报到 Qt 试试看。

至于 APNG，其实是有第三方的 Qt 图像格式插件的，我尝试过 [这个](https://github.com/Skycoder42/QtApng) 并且也修复过其中的一些问题（也已经合并到了上游）。如果有意愿的话可以试试看。由于 APNG 不算是进入了标准，于是 png 库本身也没直接支持 apng 的情况下，似乎也不能指望有太广泛的上游库的支持= =
